services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/LayeredKeyValueStorage.java

```diff
@@ -151,8 +163,20 @@ public void commit() throws StorageException {
         });
   }
 
+  @Override
+  public boolean isClosed() {
+    return parent.isClosed();
+  }
+
   @Override
   public SnappedKeyValueStorage clone() {
     return new LayeredKeyValueStorage(hashValueStore, parent);
   }
+
+  private void throwIfClosed() {
+    if (parent.isClosed()) {
+      LOG.error("Attempting to use a closed RocksDBKeyValueStorage");
+      throw new IllegalStateException("Storage has been closed");
```

Comment:
RocksDBSnapshotTransaction uses StorageException - does it make sense to match that here?
      throw new StorageException("Storage has already been closed");

Reply:
Yes, I think storage exception is better everywhere possible :+1: